account/account.go

```diff
@@ -252,3 +256,16 @@ func useGDPRChannelEnabled(account *config.Account) bool {
 func useCCPAChannelEnabled(account *config.Account) bool {
 	return account.CCPA.ChannelEnabled.IsSet() && !account.CCPA.IntegrationEnabled.IsSet()
 }
+
+// deprecateEventsEnabledField is responsible for ensuring backwards compatibility of "events_enabled" field.
+// This function favors "events.enabled" field over deprecated "events_enabled" field, if values for both are set.
+// If only deprecated "events_enabled" field is set then it sets the same value to "events.enabled" field.
+func deprecateEventsEnabledField(account *config.Account) {
+	if account != nil && account.Events.Enabled == nil {
+		account.Events.Enabled = account.EventsEnabled
+	}
+	if account != nil {
```

Comment:
In order to avoid the second check we can nest the first `if` statement:
```
263   func deprecateEventsEnabledField(account *config.Account) {
264 -     if account != nil && account.Events.Enabled == nil {
265 -         account.Events.Enabled = account.EventsEnabled
266 -     }
267       if account != nil {
    +         if account.Events.Enabled == nil {
    +             account.Events.Enabled = account.EventsEnabled
    +         }
268           // assign the old value to the new value so old and new are always the same even though the new value is what is used in the application code.
269           account.EventsEnabled = account.Events.Enabled
270       }
271   }
account/account.go
```